pkg/tfbridge/replaceErrorProperties.go

```diff
@@ -0,0 +1,40 @@
+package tfbridge
+
+import (
+	"strings"
+
+	shim "github.com/pulumi/pulumi-terraform-bridge/v3/pkg/tfshim"
+)
+
+func replaceSubstring(s, old, new string) string {
+	return strings.ReplaceAll(s, old, new)
+}
+
+func replaceSubstrings(s string, replacements map[string]string) string {
+	for k, v := range replacements {
+		s = replaceSubstring(s, k, v)
+	}
+	return s
+}
+
+func getConfigReplacements(
+	providerName string,
+	config map[string]*SchemaInfo,
+	schema shim.SchemaMap,
+) map[string]string {
+	renames := make(map[string]string)
+	schema.Range(func(key string, value shim.Schema) bool {
```

Comment:
The for-loop is not a great strategy.. If you have foo and fooProp it might replace twice. Why don't you build a regex and replace in one pass.